lib/wallet/walletdb.js

```diff
@@ -176,8 +179,17 @@ class WalletDB extends EventEmitter {
    */
 
   async open() {
+    const migrations = new WalletMigrations({
+      ...this.options,
+      walletDB: this,
+      dbVersion: this.version
+    });
+
+    const migrationResult = await migrations.migrate();
+
+    this.logger.info('Opening WalletDB...');
     await this.db.open();
-    await this.db.verify(layout.V.encode(), 'wallet', 0);
+    await this.db.verify(layout.V.encode(), this.name, this.version);
```

Comment:
Why are leaving walletdb verify simple like this but in chainDB you replaced this line entirely with get version / initialize / verifyVersion, etc?

Reply:
Chaindb logic depends on the "atomicity" of the fresh start (even though it did not have atomicity there) and difference between fresh and restarting dbs is big. In wallet however, there's no such difference and there's no need to do them separately.